cpp/src/copying/scatter_to_table.cu

```diff
@@ -0,0 +1,186 @@
+                                                            MapIterator column_labels_end,
+                                                            size_type num_output_rows,
+                                                            size_type num_output_columns,
+                                                            rmm::mr::device_memory_resource* mr,
+                                                            cudaStream_t stream) const
+  {
+    // Generate a column of all nulls:
+    std::unique_ptr<column> target_column;
+    if (input.type() == data_type{type_id::STRING}) {
+      target_column =
+        std::make_unique<column>(input.type(),
+                                 num_output_rows * num_output_columns,
+                                 rmm::device_buffer{0, stream, mr},
+                                 create_null_mask(input.size(), mask_state::ALL_NULL, stream, mr),
+                                 num_output_rows * num_output_columns);
+    } else {
+      target_column = make_fixed_width_column(
+        input.type(), num_output_columns * num_output_rows, mask_state::ALL_NULL, stream);
+    }
+
+    auto target_table = table_view{{target_column->view()}};
+
+    auto it = thrust::make_transform_iterator(
+      thrust::make_counting_iterator<size_type>(0),
+      index_pair_to_index<MapIterator>(row_labels_begin, column_labels_begin, num_output_rows));
+
+    auto result_table =
+      detail::scatter(table_view{{input}}, it, it + input.size(), target_table, false, mr, stream);
+
+    auto result_column = std::move(result_table->release()[0]);
+
+    std::vector<size_type> splits(num_output_columns - 1);
+    std::generate(splits.begin(), splits.end(), [n = 0, num_output_rows]() mutable {
+      n += num_output_rows;
+      return n;
+    });
+
+    table_view result_table_view{cudf::split(result_column->view(), splits)};
+
+    return std::make_pair(std::move(result_column), result_table_view);
+  }
+};
+
+template <typename MapIterator>
+std::pair<std::unique_ptr<column>, table_view> scatter_to_table(column_view const& input,
+                                                                MapIterator row_labels_begin,
+                                                                MapIterator row_labels_end,
+                                                                MapIterator column_labels_begin,
+                                                                MapIterator column_labels_end,
+                                                                size_type num_output_rows,
+                                                                size_type num_output_columns,
+                                                                rmm::mr::device_memory_resource* mr,
+                                                                cudaStream_t stream)
+{
+  auto scatter_to_table_functor = column_to_table_scatterer<decltype(row_labels_begin)>{};
+  return type_dispatcher(input.type(),
```

Comment:
Why do you need to dispatch on the `input` type here? When you later call `cudf::scatter` it's just going to have to do the same dispatch again.